rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py

```diff
@@ -215,33 +220,54 @@ def _lm_specific_token_cleanup(
 
         Returns: Cleaned up token ids and token strings.
         """
-        from rasa.nlu.utils.hugging_face.registry import model_tokens_cleaners
+        token_ids_string = [
+            (id_, re.sub(TOKENIZER_PREFIX_CHARS, "", token))
```

Comment:
Actually, I wonder if we can just do:
```
split_token_strings = self.tokenizer.convert_ids_to_tokens(split_token_ids, skip_special_tokens=True)
```
in `_lm_tokenize`

Reply:
I think this will only remove the special tokens (`[CLS], [UNK], etc.`), not the prefixes from the split sub tokens. To test I tried the following:

```
tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
text = "awesome word embeddings"
token_ids = tokenizer.encode(text, add_special_tokens=True)
print(token_ids)
print(tokenizer.convert_ids_to_tokens(token_ids, skip_special_tokens=False))
print(tokenizer.convert_ids_to_tokens(token_ids, skip_special_tokens=True))
```

resulting in

```
[101, 12476, 2773, 7861, 8270, 4667, 2015, 102]
['[CLS]', 'awesome', 'word', 'em', '##bed', '##ding', '##s', '[SEP]']
['awesome', 'word', 'em', '##bed', '##ding', '##s']
```

still leaving the `##` in the output strings.

Reply:
That's a shame... Thank you for looking into it!